ple/games/donkeykong/Fireball.py

```diff
@@ -0,0 +1,118 @@
+__author__ = 'Erilyth'
+import pygame
+import math
+from OnBoard import OnBoard
+
+'''
+This class defines all our fireballs.
+A fireball inherits from the OnBoard class since we will use it as an inanimate object on our board.
+Each fireball can check for collisions in order to decide when to turn and when they hit a player.
+'''
+
+class Fireball(OnBoard):
+    def __init__(self, raw_image, position, index, speed, rng):
+        super(Fireball,self).__init__(raw_image, position, rng)
+        #Set the fireball direction randomly
+        self.rng = rng
+        self.__direction = int(math.floor(self.rng.rand() * 100)) % 2
+        self.index = index
+        self.wallsBelow = []
+        self.laddersBelow = []
+
+        self.IMAGES = {
+            "fireballright": pygame.transform.scale(pygame.image.load('Assets/fireballright.png'), (15, 15)), 
```

Comment:
performance: Might be useful to use image.convert() or image.convert_alpha() depending on if the image uses alpha. 

I should have mentioned on my first pass through. You can just iterate through the dict converting & assigning it